### PR TITLE
[TypeScript SDK] Support passing a vector of objects in LocalTxnBuilder

### PR DESCRIPTION
**Pull request by 666lcz**
Closes https://github.com/MystenLabs/sui/issues/4921

### CODE REVIEW

File: sdk/typescript/src/signers/txn-data-serializers/call-arg-serializer.ts [hunk 1]
```diff
@@ -91,6 +91,24 @@ export class CallArgSerializer {
...
+      'Struct' in expectedType.Vector
+    ) {
+      if (!Array.isArray(argVal)) {
+        throw new Error(
+          `Expect ${argVal} to be a array, received ${typeof argVal}`
+        );
+      }
+      return {
+        ObjVec: await Promise.all(
+          argVal.map(async (arg) => this.newObjectArg(arg as string))
```
Comment (Jordan-Mysten) [inline sdk/typescript/src/signers/txn-data-serializers/call-arg-serializer.ts:107]: You shouldn't need to make the map function async.

File: sdk/typescript/test/e2e/object-vector.test.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  getEvents,
+  getExecutionStatusType,
+  LocalTxnDataSerializer,
+  ObjectId,
+  RawSigner,
+} from '../../src';
+import { DEFAULT_GAS_BUDGET, setup, TestToolbox } from './utils/setup';
+// TODO: this file is generated through
+// `sui move build --dump-bytecode-as-base64 --path sui/crates/sui-core/src/unit_tests/data/entry_point_vector`
+// This is fragile but needed for now because we don't know the path of user's Sui CLI in this e2e test. A
```
Comment (Jordan-Mysten) [inline sdk/typescript/test/e2e/object-vector.test.ts:17]: You should be able to use cargo to invoke the sui CLI?

**Review by Jordan-Mysten — APPROVED** [review]
Few nits but LGTM

File: sdk/typescript/test/e2e/object-vector.test.ts (hunk 2 above)
Comment (666lcz) [reply sdk/typescript/test/e2e/object-vector.test.ts]: Good point! Rewrite the logic to use `cargo run` instead. But I noticed that it takes around ~5 seconds to run the `cargo run` command(even when the binary has been built), so we may still want to add the `move build` endpoint to the `sui-test-validator`

Comment (Jordan-Mysten) [reply sdk/typescript/test/e2e/object-vector.test.ts]: Yeah I think the test validator could hold this in the future for sure!

**Pull request merged**